src/Generate.purs

```diff
@@ -146,44 +146,39 @@ INPUTS
 in {
   inherit inputs;
 
-  installSpagoStyle = pkgs.runCommand "install-spago-style" {} ''
-      >>$out echo "#!/usr/bin/env bash"
-      >>$out echo
-      >>$out echo "echo installing dependencies..."
-      >>$out echo "${builtins.toString (
-        builtins.map cpPackage (builtins.attrValues inputs))}"
-      >>$out echo "echo done."
-      chmod +x $out
+  installSpagoStyle = pkgs.writeShellScriptBin "install-spago-style" ''
+    echo "installing dependencies..."
+    ${builtins.toString (
+       builtins.map cpPackage (builtins.attrValues inputs))}
+    echo "done."
   '';
 
-  buildSpagoStyle = pkgs.runCommand "build-spago-style" {} ''
-      >>$out echo "#!/usr/bin/env bash"
-      >>$out echo
-      >>$out echo "echo building project..."
-      >>$out echo "purs compile ${builtins.toString (
-        builtins.map getGlob (builtins.attrValues inputs))}" \"\$@\"
-      >>$out echo "echo done."
-      chmod +x $out
+  buildSpagoStyle = pkgs.writeShellScriptBin "build-spago-style" ''
+    EXTRA_FILES=$@
+
+    echo "echo building project..."
+    echo "purs compile ${builtins.toString (
+      builtins.map getGlob (builtins.attrValues inputs))}" \"\$EXTRA_FILES\"
+    echo "echo done."
   '';
 
-  buildFromNixStore = pkgs.runCommand "build-from-store" {} ''
-      >>$out echo "#!/usr/bin/env bash"
-      >>$out echo
-      >>$out echo "echo building project using sources from nix store..."
-      >>$out echo "purs compile ${builtins.toString (
-        builtins.map getStoreGlob (builtins.attrValues inputs))}" \"\$@\"
-      >>$out echo "echo done."
-      chmod +x $out
+  buildFromNixStore = pkgs.writeShellScriptBin "build-from-store" ''
+    EXTRA_FILES=$@
+
+    echo "echo building project using sources from nix store..."
+    echo "purs compile ${builtins.toString (
+      builtins.map getStoreGlob (builtins.attrValues inputs))}" \"\$EXTRA_FILES\"
+    echo "echo done."
   '';
 
   mkBuildProjectOutput =
-    { src, purs }:
+    { src, purescript }:
```

Comment:
Isn't it better to depend on purescript, as that's the package that contains the `purs` binary?

Reply:
sure, it's just a name change

Reply:
see https://github.com/justinwoo/easy-purescript-nix/blob/054caeb3cd4292920e21a4de92a9decbac8e8953/default.nix#L35-L39